spec/models/variant_price_spec.rb

```diff
@@ -105,4 +105,114 @@
       end
     end
   end
+
+  describe "tier-specific duration functionality" do
+    let(:variant) { create(:variant) }
+    let(:variant_price) { create(:variant_price, variant: variant, recurrence: "monthly") }
```

Comment:
Please try using non-persisted records for testing simple instance methods as it spares the test suite the round trips to the database. Our test suite is large, we need to avoid making it slower than it needs to be.

You can use `build` instead of `create` and you can use `assign_attributes` instead of `update!`